src/codeu/chat/server/Server.java

```diff
@@ -66,13 +76,90 @@
   private final Relay relay;
   private Uuid lastSeen = Uuid.NULL;
 
+  private void loadState() {
+
+    File file = new File(FileWriter.TRANSACTION_FILE);
+
+    // check if can't load transaction state from the log file
+    if(!file.exists())
+      return;
+    FileInputStream fin = null;
+    try {
+      fin = new FileInputStream(file);
+    } catch (FileNotFoundException e) {
+      System.err.println("couldn't find transaction log file");
+    } catch (SecurityException e) {
+      System.err.println("can't access read transaction log file");
+    }
+
+    // each count has a type first then data right after
+    String type;
+    Object value;
+    try {
+
+      // each object is in the format: 0x00 then type then data
+      while(fin.read() != -1) {
+
+          // read the type of the data
+          type = Serializers.STRING.read(fin);
+
+          // check cases for user message or conversationheader types
+          switch(type) {
+            case Writeable.USER_STR:
+              value = User.SERIALIZER.read(fin);
+              User user = (User)value;
+
+              // add new user to restore state
+              this.controller.newUser(user.id, user.name, user.creation);
+              break;
+            case Writeable.MESSAGE_STR:
+              value = Message.SERIALIZER.read(fin);
+              Message message = (Message)value;
+
+              // add new message to restore state
+              this.controller.newMessage(message.id, message.author, message.next, message.content, message.creation);
+              break;
+            case Writeable.CONVERSATION_STR:
+              value = ConversationHeader.SERIALIZER.read(fin);
+              ConversationHeader conversationheader = (ConversationHeader)value;
+
+              // add new conversation to restore state
+              this.controller.newConversation(conversationheader.id, conversationheader.title, conversationheader.owner, conversationheader.creation);
+              break;
+            }
+      }
+    } catch (IOException e) {
+        System.err.println("error reading transaction log");
+      }
+
+    try {
+      fin.close();
+    } catch (IOException e) {
+    }
+
+  }
+
   public Server(final Uuid id, final Secret secret, final Relay relay) {
 
     this.id = id;
     this.secret = secret;
-    this.controller = new Controller(id, model);
+
+    BlockingQueue<Writeable> blockq = new LinkedBlockingQueue<Writeable>();
+    FileWriter filewriter = new FileWriter(blockq);
+
+    this.controller = new Controller(id, model, filewriter);
     this.relay = relay;
 
+    // new controller default to not save
+    // load the state of the server on startup
+    loadState();
+
+    // after loading, can start saving to log files when server starts
+    this.controller.setSave(true);
```

Comment:
This still seems messy. If you made controller entirely responsible for loading and writing transactions, you wouldn't need this out here. Instead, you could initiate loading transactions from the controller, as well as instantiating the transaction writing thread. That way, the Server doesn't need to know the particular implementation details of the Controller (this is sometimes called leaking abstraction).

In the grander picture, the controller itself would delegate loading to another class (the above suggested TransactionFileReader), and also delegate the writing. Making code modular like this improves maintainability (as separating concerns makes it easier to change one thing without breaking a whole host of other things) and makes it easier to unit test.